### DISCUSSION TITLE
byte ordering of ipv4_t and KENS

### DISCUSSION THREAD

**Discussion by livelykitten** [ANLAB-KAIST/KENSv3#160] [Project Q&A 2. Transport Layer (TCP)] [answered]
What is the byte order of ipv4_t? does KENS use network byte order for mac address, ip address, port number and other values?

Thank you

**Comment by Pusnow** [comment]
Yes, you should follow network byte order for all values in TCP/IP headers.

**Reply by livelykitten** [reply]
Does that mean ipv4_t stores MSB of ip address in index 0 of its array?

e.g. for 127.0.0.1, 127 stored in array index 0?

**Reply by Pusnow** [reply, marked as answer]
Yes.
